tests/Library/Vanilla/VisibleTextLengthValidatorTest.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * @copyright 2009-2019 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+namespace VanillaTests\Library\Vanilla;
+
+use Vanilla\Invalid;
+use Vanilla\VisibleTextLengthValidator;
+use Vanilla\Formatting\FormatService;
+use VanillaTests\MinimalContainerTestCase;
+
+/**
+ * Tests for VisibleTextLengthValidator class.
+ *
+ * Class VisibleTextLengthValidatorTest
+ * @package VanillaTests\Library\Vanilla
+ */
+class VisibleTextLengthValidatorTest extends MinimalContainerTestCase {
+
+    /** @var int Number of characters to test for. */
+    const TEST_MAX_COMMENT_LENGTH = 10;
+
+    /**
+     * Test the VisibleTextLengthValidator Class
+     *
+     * @param array $post Testable POST array.
+     * @param bool $isValid Expected result of the validator.
+     *
+     * @dataProvider providePostContent()
+     */
+    public function testVisibleTextLengthValidator(array $post, bool $isValid) {
+        $formatService = \Gdn::formatService();
+        $locale = \Gdn::locale();
+        $validator = new VisibleTextLengthValidator(self::TEST_MAX_COMMENT_LENGTH, $formatService, $locale);
+        $result = $validator($post['Body'] ?? '', 'Body', $post);
+        $resultType = true;
+        if ($result instanceof Invalid) {
+            $resultType = false;
+        }
+        $this->assertEquals($isValid, $resultType);
```

Comment:
You could probably smoosh this all down into a couple lines. Maybe something like:

```php
$actual = !($result instanceof Invalid);
$this->assertEquals($isValid, $actual);
```